bcachefs/bcachefs.c

```diff
@@ -369,10 +451,8 @@ int Bcachefs_fini(Bcachefs *this)
 
 int Bcachefs_open(Bcachefs *this, const char *path)
 {
-    if (!Bcachefs_close(this))
-    {
-        return 0;
-    }
+    *this = (Bcachefs){0};
+
```

Comment:
In this particular case, this really looks like a side effect as a Bcachefs could be open, closed then reopen. In the same spirit as [this previous discussion](https://github.com/mila-iqia/bcachefs/pull/12#discussion_r732019157), a check could be made or we should change the signature to instead return the pointer to the freshly `malloc`ed pointer

Reply:
Or we rename this to `..._reopen` and change the behaviour to something similar to `realloc`

Reply:
But I agree that this func name is misleading.